src/main/java/org/killbill/billing/plugin/adyen/dao/AdyenDao.java

```diff
@@ -17,556 +17,402 @@
+            return DSL.using(conn, dialect, settings)
+                .selectFrom(ADYEN_RESPONSES)
+                .where(DSL.field(ADYEN_RESPONSES.KB_PAYMENT_ID).equal(kbPaymentId.toString()))
+                .and(DSL.field(ADYEN_RESPONSES.KB_TENANT_ID).equal(kbTenantId.toString()))
+                //                .and(DSL.field(ADYEN_RESPONSES.TRANSACTION_TYPE).equal(PURCHASE))
+                .orderBy(ADYEN_RESPONSES.RECORD_ID)
+                .fetchOne();
+          }
+        });
+  }
+
+  public AdyenResponsesRecord getResponseFromMerchantReference(final String merchantReference)
+      throws SQLException {
+    return execute(
+        dataSource.getConnection(),
+        new WithConnectionCallback<AdyenResponsesRecord>() {
+          @Override
+          public AdyenResponsesRecord withConnection(final Connection conn) throws SQLException {
+            return DSL.using(conn, dialect, settings)
+                .selectFrom(ADYEN_RESPONSES)
+                .where(
+                    DSL.field(ADYEN_RESPONSES.KB_PAYMENT_TRANSACTION_ID).equal(merchantReference))
+                .orderBy(ADYEN_RESPONSES.RECORD_ID)
+                .fetchOne();
+          }
+        });
+  }
+
+  public AdyenPaymentMethodsRecord getPaymentMethod(final String kbPaymentMethodId)
+      throws SQLException {
+    return execute(
+        dataSource.getConnection(),
+        new WithConnectionCallback<AdyenPaymentMethodsRecord>() {
+          @Override
+          public AdyenPaymentMethodsRecord withConnection(final Connection conn)
+              throws SQLException {
+            return DSL.using(conn, dialect, settings)
+                .selectFrom(ADYEN_PAYMENT_METHODS)
+                .where(
+                    DSL.field(ADYEN_PAYMENT_METHODS.KB_PAYMENT_METHOD_ID).equal(kbPaymentMethodId))
+                .fetchOne();
+          }
+        });
+  }
+
+  public List<AdyenResponsesRecord> getSuccessfulPurchaseResponseList(
+      final UUID kbPaymentId, final UUID kbTenantId) throws SQLException {
+    return execute(
+        dataSource.getConnection(),
+        new WithConnectionCallback<List<AdyenResponsesRecord>>() {
+          @Override
+          public List<AdyenResponsesRecord> withConnection(final Connection conn)
+              throws SQLException {
+            return DSL.using(conn, dialect, settings)
+                .selectFrom(ADYEN_RESPONSES)
+                .where(DSL.field(ADYEN_RESPONSES.KB_PAYMENT_ID).equal(kbPaymentId.toString()))
+                .and(DSL.field(ADYEN_RESPONSES.KB_TENANT_ID).equal(kbTenantId.toString()))
+                .orderBy(ADYEN_RESPONSES.RECORD_ID.desc())
+                .fetch();
+          }
+        });
+  }
+
+  @SuppressWarnings("rawtypes")
+  public static Map mapFromAdditionalDataString(@Nullable final String additionalData) {
+    if (additionalData == null) {
+      return ImmutableMap.of();
```

Comment:
use `Collections.emptyMap()` (preferred, as consistent across our code base), or `Map.of()` (shorter).